source/sas_compress.cpp

```diff
@@ -280,24 +283,46 @@ std::string LZ4Compressor::compress(const std::string& s, const SAS::Profile* pr
       _saved_streams[profile] = saved_lz4_stream(base_stream, stream_saved_buf);
       saved_stream_iterator = _saved_streams.find(profile);
     }
-    
-    LZ4_stream_restore_preserved(_stream,
-                                 saved_stream_iterator->second.first,
-                                 saved_stream_iterator->second.second);
   }
 
+
+
```

Comment:
Yes, nice, like this a lot - maybe add a few more to really break this function up?